### ISSUE TITLE
When updating, AFO doesn't take meta-fields into account

### ISSUE THREAD

**Issue by larssn** [adriancarriger/angularfire2-offline#34]
Angularfire2 attaches metadata to object data, described here:
https://github.com/angular/angularfire2/blob/master/docs/2-retrieving-data-as-objects.md#meta-fields-on-the-object

AFO doesn't seem to take these into account when you update an object, and fails with the message:
`Firebase.update failed: First argument contains a function in property 'users.8xfSTpTysPRY8gCKVyKZCGZJ1wg2.$exists' with contents: function () { return _this.value !== null; }`

Seems its trying to write the function as data.

**Comment by adriancarriger** [comment]
I haven't seen this error before. How can I reproduce it? It would be awesome if you can post all your relevant code. Thanks!

**Comment by larssn** [comment]
It shouldn't be too hard to reproduce.

I'm gonna use plain angularfire2 in my example, but you can swap in AfoObjectObservable to see the problem.

```
@Component({
    template: '
<ion-content>
    <ion-item>
      <ion-label for="firstname" floating>First name</ion-label>
      <ion-input [(ngModel)]="customerData.first_name" (change)="save()"></ion-input>
    </ion-item>

    <ion-item>
      <ion-label floating>Last name</ion-label>
      <ion-input [(ngModel)]="customerData.last_name" (change)="save()"></ion-input>
    </ion-item>
</ion-content>'
})
export class ProfileComponent implements OnInit {

    customerData = {};
    customer: FirebaseObjectObservable<User>;

    constructor(
        private auth: AngularFireAuth,
        private db: AngularFireDatabase
    ) { }

    ngOnInit() {
        const sub = this.auth.authState
            .subscribe(user => {
                if (user && !user.isAnonymous) {
                    this.customer = this.db.object(`users/${user.uid}`);
                    this.customer.subscribe(data => {
                        this.customerData = data;
                    });
                    sub.unsubscribe();
                }
            });
    }

    save() {
        this.customer.update(this.customerData);
    }
}
```

The problem here is that `customerData` will contain meta fields we don't want our updated data to contain. They're described in the link in the OP, and it also contains a function (that I cant find docs on atm) called `$exists()`. AFO will fail because it tries to write this function as data. AF2 handles this automatically, it ignores the meta fields when updating.

Have you considered extending AF2's functionality? Then these issues would be less likely. :)

Keep up the good work!

**Comment by adriancarriger** [comment]
@larssn Thanks for the update! I was able to reproduce your [issue here](https://github.com/adriancarriger/angularfire2-offline/blob/gh-pages/issues/34.1/src/app/app.component.ts) ([live demo](https://adriancarriger.github.io/angularfire2-offline/issues/34.1/dist/)).

It doesn't look too difficult so hopefully it will be solved soon! 👍

**Comment by adriancarriger** [comment]
This is now fixed in version 4.1.3 🎉 

Here is a [demo](https://adriancarriger.github.io/angularfire2-offline/issues/34.2/dist/) (with [code](https://github.com/adriancarriger/angularfire2-offline/blob/gh-pages/issues/34.2/src/app/app.component.ts)).

**Comment by larssn** [comment]
All these demos 😄 , you have too much time on your hands mate! 😆 

Anyway, thanks for the snappy fix!
